prov/gni/src/gnix_vc.c

```diff
@@ -1944,6 +1945,13 @@ static int __gnix_vc_push_work_reqs(struct gnix_vc *vc)
 				GNIX_WARN(FI_LOG_EP_DATA,
 					  "Failed to push request %p: %s\n",
 					  req, fi_strerror(-ret));
+
+				/*
+				 * If a UGNI fn returned invalid param, quit.
+				 */
+				if (ret == -FI_EINVAL) {
+					GNIX_FATAL(FI_LOG_EP_DATA, "\n");
+				}
```

Comment:
> Do you want to just change the exiting GNIX_WARN right above to GNIX_FATAL?

I think that's best.  The program spews out a lot of warnings before crashing when a non-zero return value is received here.